### PR TITLE
rgw: Correct the return codes for the health check feature

### PR DESCRIPTION
**Pull request by prallabh**
Fixes: http://tracker.ceph.com/issues/19025
Signed-off-by: Pavan Rallabhandi <PRallabhandi@walmartlabs.com>

### CODE REVIEW

**Comment by cbodley** [comment]
looks correct to me. what do you think @rzarzynski ?

**Comment by prallabh** [comment]
@rzarzynski can you please take a look

File: src/rgw/rgw_op.cc [hunk 1]
```diff
@@ -5202,9 +5202,13 @@ void RGWListBucketMultiparts::execute()
...
 void RGWGetHealthCheck::execute()
 {
-  if (! g_conf->rgw_healthcheck_disabling_path.empty() &&
-      ::access(g_conf->rgw_healthcheck_disabling_path.c_str(), F_OK )) {
-    op_ret = -ERR_SERVICE_UNAVAILABLE;
+  if (!g_conf->rgw_healthcheck_disabling_path.empty()) {
+    if (::access(g_conf->rgw_healthcheck_disabling_path.c_str(), F_OK) == 0) {
+      /* path exists */
+      op_ret = -ERR_SERVICE_UNAVAILABLE;
+    } else {
```
Comment (rzarzynski) [inline src/rgw/rgw_op.cc]: Is there a reason for duplicating the `200 OK` case? Maybe something like below?
```
  if (! g_conf->rgw_healthcheck_disabling_path.empty()) &&
      ::access(g_conf->rgw_healthcheck_disabling_path.c_str(), F_OK ) == 0) {
    /* Disabling path specified & existent in the filesystem. */
    op_ret = -ERR_SERVICE_UNAVAILABLE; /* 503 */
  } else {
    op_ret = 0; /* 200 OK */
  }
```

Comment (prallabh) [inline src/rgw/rgw_op.cc]: No reason but for to be more explicit on the path existence and access failures. Corrected it now.

Comment (rzarzynski) [inline src/rgw/rgw_op.cc]: Thanks!

**Comment by rzarzynski** [comment]
jenkins test this please

**Comment by rzarzynski** [comment]
jenkins test this please (unrelated failure in `ceph_objectstore_tool.py`)

**Pull request merged**